auth/session_guard.go

```diff
@@ -0,0 +1,142 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/spf13/cast"
+
+	contractsauth "github.com/goravel/framework/contracts/auth"
+	"github.com/goravel/framework/contracts/cache"
+	"github.com/goravel/framework/contracts/config"
+	"github.com/goravel/framework/contracts/http"
+	contractsession "github.com/goravel/framework/contracts/session"
+	"github.com/goravel/framework/errors"
+)
+
+type SessionGuard struct {
+	cache    cache.Cache
+	config   config.Config
+	session  contractsession.Session
+	ctx      http.Context
+	guard    string
+	provider contractsauth.UserProvider
+}
+
+func NewSessionGuard(ctx http.Context, name string, userProvider contractsauth.UserProvider) (contractsauth.GuardDriver, error) {
+	if ctx == nil {
+		return nil, errors.InvalidHttpContext.SetModule(errors.ModuleAuth)
+	}
+	if cacheFacade == nil {
+		return nil, errors.CacheFacadeNotSet.SetModule(errors.ModuleAuth)
+	}
+	session := ctx.Request().Session()
+	if session == nil {
+		return nil, errors.SessionDriverIsNotSet.SetModule(errors.ModuleAuth)
+	}
+
+	return &SessionGuard{
+		cache:    cacheFacade,
+		config:   configFacade,
+		session:  session,
+		ctx:      ctx,
+		guard:    name,
+		provider: userProvider,
+	}, nil
+}
+
+// Check implements auth.GuardDriver.
+func (r *SessionGuard) Check() bool {
+	_, err := r.ID()
+
+	if err == nil {
+		return true
+	}
+
+	return false
+}
+
+// Guest implements auth.GuardDriver.
+func (r *SessionGuard) Guest() bool {
+	return !r.Check()
+}
+
+// ID implements auth.GuardDriver.
+func (r *SessionGuard) ID() (token string, err error) {
+	sessionName := r.getSessionName()
+	userId := r.session.Get(sessionName, nil)
```

Comment:
We always use `*ID` in the framework.

```suggestion
	userID := r.session.Get(sessionName, nil)
```